server/service/src/ledger_fix/ledger_fix_driver.rs

```diff
@@ -0,0 +1,134 @@
+use std::sync::Arc;
+
+use crate::{activity_log::system_error_log, service_provider::ServiceProvider};
+
+use chrono::{NaiveDateTime, TimeDelta, Utc};
+use repository::system_log_row::SystemLogType;
+use repository::{KeyType, KeyValueStoreRepository, RepositoryError};
+use tokio::{
+    sync::mpsc::{self, Receiver, Sender},
+    time::Duration,
+};
+
+pub struct LedgerFixDriver {
+    receiver: Receiver<Option<String>>,
+}
+
+#[derive(Clone)]
+pub struct LedgerFixTrigger {
+    sender: Sender<Option<String>>,
+}
+
+const FIRST_RUN_DELAY: Duration = Duration::from_secs(5);
+// This trigger is not to re-run ledger fix but to check if ledger fix is needed again
+// Will check against key values store LAST_LEDGER_FIX_RUN and LEDGER_FIX_INTERVAL
+const RE_TRIGGER_DELAY: Duration = Duration::from_secs(60 * 60); // 1 hour
+const LEDGER_FIX_INTERVAL: TimeDelta = TimeDelta::days(1);
```

Comment:
Was having some thoughts about trying to run in off hours (say 12am) but particularly smaller sites may turn their desktops or tablets off over night